core/trino-main/src/main/java/io/trino/server/remotetask/HttpRemoteTask.java

```diff
@@ -998,6 +998,32 @@ public void onFailure(Throwable t)
                     errorScheduledExecutor.schedule(() -> doScheduleAsyncCleanupRequest(cleanupBackoff, request, action), delayNanos, NANOSECONDS);
                 }
             }
+
+            private void fatalAsyncCleanupFailure(Throwable cause)
+            {
+                synchronized (HttpRemoteTask.this) {
```

Comment:
@pettyjamesm do we need this `synchronized` I followed to general pattern but it is not obvious it is needed to me. 
Is that we should only call `updateTaskInfo` when synchronized on `HttpRemoteTask`? If so it means we should also add `synchronized` to `FutureCallback#onSuccess` used in `doScheduleAsyncCleanupRequest`.

Or maybe we can just drop it from `fatalAsyncCleanupFailure` and `fatalUnacknowledgedFailure`?

Reply:
Honestly, I'm not 100% sure about all the concurrency guarantees `HttpRemoteTask` is actually attempting to enforce (other than the obvious task status / split assignment logic). It would seem like any attempt to update `TaskStatus` after checking the current value is probably a candidate for mutual exclusion, but most situations appear to be handled by other synchronizations. I would leave it here out of caution, even though this particular call shouldn't really be racing with other operations.